computer_vision/richardson_lucy.py

```diff
@@ -0,0 +1,241 @@
+        data_type: numpy data type to set output variable to
+    Returns:
+        return 2D numpy array of type uint8, corresponding to limited range matrix
+
+    Examples:
+        >>> normalize_image(np.array([[1, 2, 3], [4, 5, 10]]),
+        ...                 cap=1.0, data_type=np.float64)
+        array([[0.        , 0.11111111, 0.22222222],
+               [0.33333333, 0.44444444, 1.        ]])
+        >>> normalize_image(np.array([[4, 4, 3], [1, 7, 2]]))
+        array([[127, 127,  85],
+               [  0, 255,  42]], dtype=uint8)
+    """
+    normalized = (image - np.min(image)) / (np.max(image) - np.min(image)) * cap
+    return normalized.astype(data_type)
+
+
+def gaussian_noise(size: tuple, mean=0, std=0.05):
+    """Creates normal distribution array with given size to use as noise.
+
+    Args:
+        size: Size of the desired output image
+        mean: Mean to use within the Gaussian Function
+        std: Standard deviation to use within the Gaussian Function
+
+    Returns:
+        Matrix with given size, containing generated gaussian values.
+
+    Example:
+        >>> np.random.seed(0)
+        >>> gaussian_noise((5, 5))
+        array([[ 22.49166741,   5.10200441,  12.4789093 ,  28.57138829,
+                 23.81136437],
+               [-12.46029297,  12.11362732,  -1.92980441,  -1.31604036,
+                  5.2351309 ],
+               [  1.83655553,  18.54198721,   9.703231  ,   1.55135646,
+                  5.65925622],
+               [  4.25434767,  19.04950818,  -2.61576786,   3.9916132 ,
+                -10.88972068],
+               [-32.55062015,   8.33363709,  11.02156154,  -9.46260401,
+                 28.93937146]])
+    """
+    noise = np.multiply(np.random.normal(mean, std, size), 255)
+    return noise
+
+
+def gaussian_filter(k: int = 5, sigma: float = 1.0) -> np.ndarray:
+    """Generates a matrix with weights corresponding to centered gaussian distribution.
+
+    Args:
+        k: Lateral size of the kernel.
+        sigma: Standard deviation to be used when generating distribution
+
+    Returns:
+        np.ndarray: [k x k] sized kernel to be used as filter
+
+    Examples:
+        >>> gaussian_filter()
+        array([[0.00296902, 0.01330621, 0.02193823, 0.01330621, 0.00296902],
+               [0.01330621, 0.0596343 , 0.09832033, 0.0596343 , 0.01330621],
+               [0.02193823, 0.09832033, 0.16210282, 0.09832033, 0.02193823],
+               [0.01330621, 0.0596343 , 0.09832033, 0.0596343 , 0.01330621],
+               [0.00296902, 0.01330621, 0.02193823, 0.01330621, 0.00296902]])
+    """
+    arx = np.arange((-k // 2) + 1.0, (k // 2) + 1.0)
+    x, y = np.meshgrid(arx, arx)
+    filt = np.exp(-(1 / 2) * (np.square(x) + np.square(y)) / np.square(sigma))
+    return filt / np.sum(filt)
+
+
+def convolve(matrix: np.ndarray, kernel: np.ndarray) -> np.ndarray:
+    """
+    Convolves a given kernel around a matrix through the frequency domain,
+    using Fourier transformations.
```

Comment:
```suggestion
    using Fourier transforms.
```
Nitpick: "Fourier transforms", not "Fourier transformations"